From e5930f27980fe0004c1c5f97634bd30f31d07043 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 5 Jan 2017 17:00:37 +0800
Subject: [PATCH 1457/5242] MLK-13722-3 video: mxsfb: add more sanity check on
 overlayfb_check_var()

commit  9e0e2e9583d851360726d2f3ff62db514f3c0657 from
https://source.codeaurora.org/external/imx/linux-imx.git

The overlay function of LCDIF only support when the bpp of fb0
and fb1 are the same. So add this check on overlayfb_check_var().

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 6a6134a..8530330 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1713,6 +1713,10 @@ static int overlayfb_check_var(struct fb_var_screeninfo *var,
 	struct mxsfb_info *fbi  = ofb->fbi;
 	const struct fb_bitfield *rgb = NULL;
 
+	/* lcdif doesn't support different bpp of AS and PS */
+	if (var->bits_per_pixel != fbi->fb_info->var.bits_per_pixel)
+		return -EINVAL;
+
 	/* overlay width & should be equal to fb0 */
 	if ((var->xres != fbi->fb_info->var.xres) ||
 	    (var->yres != fbi->fb_info->var.yres))
-- 
1.7.9.5

